apps/wallet/src/ui/app/pages/home/hidden-assets/index.tsx

```diff
@@ -0,0 +1,233 @@
+import PageTitle from '_src/ui/app/shared/PageTitle';
+
+const HIDDEN_ASSET_IDS = 'hidden-asset-ids';
+
+const extractName = (display?: Record<string, string> | null) => {
+	if (!display || !('name' in display)) return undefined;
+	const name = display.name;
+	if (typeof name === 'string') {
+		return name;
+	}
+	return null;
+};
+
+function NftsPage() {
+	const [internalHiddenAssetIds, internalSetHiddenAssetIds] = useState<string[]>([]);
+
+	const { data, isInitialLoading, isLoading, isError, error } = useMultiGetObjects(
+		// Prevents dupes
+		Array.from(new Set(internalHiddenAssetIds))!,
+		{ showContent: true, showDisplay: true },
+	);
+
+	const filteredAndSortedNfts = useMemo(() => {
+		const hiddenNfts = data?.flatMap((data) => data.data) || [];
+		return hiddenNfts
+			?.filter((nft) => nft && internalHiddenAssetIds.includes(nft.objectId))
+			.sort((nftA, nftB) => {
+				let nameA = '';
+				let nameB = '';
+				if (nftA && typeof nftA.display?.data !== 'string') {
+					nameA = (nftA && extractName(nftA.display?.data)) || '';
+				} else if (nftA && typeof nftA.display?.data === 'string') {
+					nameA = nftA.display?.data;
+				}
+
+				if (nftB && typeof nftB.display?.data !== 'string') {
+					nameB = (nftB && extractName(nftB.display?.data)) || '';
+				} else if (nftB && typeof nftB.display?.data === 'string') {
+					nameB = nftB.display?.data;
+				}
+
+				if (nameA < nameB) {
+					return -1;
+				} else if (nameA > nameB) {
+					return 1;
+				}
+				return 0;
+			});
+	}, [internalHiddenAssetIds, data]);
+
+	useEffect(() => {
+		(async () => {
+			const hiddenAssets = await get<string[]>(HIDDEN_ASSET_IDS);
+			if (hiddenAssets) {
+				internalSetHiddenAssetIds(hiddenAssets);
+			}
+		})();
+	}, []);
+
+	const showAssetId = useCallback(
+		async (newAssetId: string) => {
+			if (!internalHiddenAssetIds.includes(newAssetId)) return;
+
+			try {
+				const updatedHiddenAssetIds = internalHiddenAssetIds.filter((id) => id !== newAssetId);
+				internalSetHiddenAssetIds(updatedHiddenAssetIds);
+				await set(HIDDEN_ASSET_IDS, updatedHiddenAssetIds);
+			} catch (error) {
+				// Handle any error that occurred during the unhide process
+				toast.error('Failed to show asset.');
+				// Restore the asset ID back to the hidden asset IDs list
+				internalSetHiddenAssetIds([...internalHiddenAssetIds, newAssetId]);
+				await set(HIDDEN_ASSET_IDS, internalHiddenAssetIds);
+			}
+
+			const undoShowAsset = async (assetId: string) => {
+				let newHiddenAssetIds;
+				internalSetHiddenAssetIds((prevIds) => {
+					return (newHiddenAssetIds = [...prevIds, assetId]);
+				});
+				await set(HIDDEN_ASSET_IDS, newHiddenAssetIds);
+			};
+
+			const assetShownToast = async (objectId: string) => {
+				toast.custom(
+					(t) => (
+						<div
+							className="flex items-center justify-between gap-2 bg-white w-full shadow-notification border-solid border-gray-45 rounded-full px-3 py-2"
+							style={{
```

Comment:
does this need to be in a `style` prop?

Reply:
the custom toast doesn't seem to allow for the animations to be passed through the classnames unfortunately